src/ucp/core/ucp_worker.h

```diff
@@ -196,6 +196,15 @@ typedef khash_t(ucp_worker_rkey_config) ucp_worker_rkey_config_hash_t;
 KHASH_TYPE(ucp_worker_discard_uct_ep_hash, uct_ep_h, ucp_request_t*);
 typedef khash_t(ucp_worker_discard_uct_ep_hash) ucp_worker_discard_uct_ep_hash_t;
 
+typedef struct ucp_worker_mpool_key {
+    ucs_memory_type_t             mem_type;  /* memory type of the buffer pool */
+    ucs_sys_device_t              sys_dev;   /* identifier for the device,
+                                                UINT_MAX for default device */
+} ucp_worker_mpool_key_t;
+
+/* Hash map to find mpool by mpool key */
+KHASH_TYPE(ucp_worker_mpool_hash, ucp_worker_mpool_key_t, ucs_mpool_t*);
```

Comment:
mpool_t can be value instead of pointer - will simplify code

Reply:
Can you elaborate if you don't mind?

Reply:
`KHASH_TYPE(ucp_worker_mpool_hash, ucp_worker_mpool_key_t, ucs_mpool_t)`